lib/daru/io/io.rb

```diff
@@ -202,23 +200,16 @@ def from_csv_prepare_opts opts
 
       def from_csv_hash_with_headers(path, opts)
         opts[:header_converters] ||= :symbol
-
         ::CSV
-          .read(path, 'rb',opts)
-          .tap { |c| yield c if block_given? }
+          .parse(open(path), opts)
           .by_col.map { |col_name, values| [col_name, values] }.to_h
       end
 
       def from_csv_hash(path, opts)
         csv_as_arrays =
-          ::CSV
-          .open(path, 'rb', opts)
-          .tap { |c| yield c if block_given? }
```

Comment:
If you remove the `yield` the user loses the functionality to manipulate rows with some provided block right?

So something like:
``` ruby
DataFrame.from_csv('file.csv') do |row|
  row[0] = row[0].to_i
end
```

Will not work..